File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for some text.
    string text = get_string("Text:\n");

    // Count the number of letters, words, and sentences in the text.
    if (!isspace(text[0]) && ispunct(text[strlen(text) - 1]))
    {
        int letters = count_letters(text);
        int words = count_words(text);
        int sentences = count_sentences(text);

        // Compute the Coleman-Liau index.
        printf("The number of letters is: %i \n", letters);
        printf("The number of words is: %i \n", words);
        printf("The number of sentences is: %i \n", sentences);
        // Print the grade index.
        float L = (float) letters / (float) words * 100;
        float S = (float) sentences / (float) words * 100;
        int index = round(0.0588 * L - 0.296 * S - 15.8);
        if (index < 1)
        {
            printf("Before Grade 1\n");
        }
        else if (index > 16)
        {
            printf("Grade 16+\n");
        }
        else
        {
            printf("Grade %i\n", index);
        }
    }
    else
    {
        printf("Please enter a paragraph that does not begin with one or more spaces, a tab, or a new line, and that ends with a "
               "period.\n");
    }
}

int count_letters(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            count += 1;
        }
    }
    return count;
}

int count_words(string text)
{
    int count = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isspace(text[i]))
        {
            count += 1;
        }
    }
    return count;
}

int count_sentences(string text)
{
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; ++i)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            count += 1;
        }
    }
    return count;
}

File: sentimental-readability/readability.py
from cs50 import get_string
import math
# Recall that the Coleman-Liau index is computed as 0.0588 * L - 0.296 * S - 15.8, where L is the average number of letters per 100 words in the text, and S is the average number of sentences per 100 words in the text.
# Use get_string from the CS50 Library to get the user’s input, and print to output your answer.
# Your program should count the number of letters, words, and sentences in the text. You may assume that a letter is any lowercase character from a to z or any uppercase character from A to Z, any sequence of characters separated by spaces should count as a word, and that any occurrence of a period, exclamation point, or question mark indicates the end of a sentence.
# Your program should print as output "Grade X" where X is the grade level computed by the Coleman-Liau formula, rounded to the nearest integer.
# If the resulting index number is 16 or higher(equivalent to or greater than a senior undergraduate reading level), your program should output "Grade 16+" instead of giving the exact index number. If the index number is less than 1, your program should output "Before Grade 1".

def main():
  text = get_string("Text: ")
  L = countChars(text) / countWords(text) * 100
  S = countSentences(text) / countWords(text) * 100
  index = round(0.0588 * L - 0.296 * S - 15.8)
  if (index < 1):
    print("Before Grade 1")
  elif (index > 16):
    print("Grade 16+")
  else:
    result = f"Grade {index}"
    print(result)


def countWords(text):
  words = text.strip().split(" ")
  return len(words)

def countChars(text):
  chars = len(text) - text.count(" ") - text.count(",") - \
      text.count(".") - text.count("!") - text.count("?") - text.count("'")
  return chars

def countSentences(text):
  sentences = text.count(".") + text.count("!") + text.count("?")
  return sentences

main()
